.cursor/rules/makefile-commands.mdc

```diff
@@ -0,0 +1,193 @@
+---
+description: Guidelines for using Makefile commands in WAVS projects
+globs: Makefile,*.mk
+---
+# Makefile Commands Guide
+
+Essential commands for developing and deploying WAVS components.
+
+## Initial Setup
+
+1. System Requirements:
+- Docker and Docker Compose
+- Make
+- JQ
+- Node.js v21+
+- Rust v1.84+
+
+2. Mac ARM Setup (if applicable):
+```bash
+# Set up Rosetta
+softwareupdate --install-rosetta
+
+# Enable Rosetta in Docker Desktop:
+# Settings -> General -> "Use Rosetta for x86_64/amd64 emulation on Apple Silicon"
+
+# Configure networking (choose one):
+# Option 1: Docker Desktop: Settings -> Resources -> Network -> 'Enable Host Networking'
+# Option 2: Run this command:
+brew install chipmk/tap/docker-mac-net-connect && sudo brew services start chipmk/tap/docker-mac-net-connect
+```
+
+## Core Build Commands
+
+1. Initial Setup:
+```bash
+# Install initial dependencies (npm packages & submodules)
+make setup
+
+# Build entire project (Forge contracts + WASI components)
+make build
+
+# Build only WASI components
+make wasi-build
+# Optionally specify directory:
+WASI_BUILD_DIR=components/your-component make wasi-build
+```
+
+## Complete Deployment Sequence
+
+1. Start Local Environment:
+```bash
+# Start anvil (local Ethereum node) and WAVS services
+# This must remain running in your terminal
+make start-all
+```
+
+2. Deploy Service Contracts:
+```bash
+# Get Eigenlayer service manager address
+export SERVICE_MANAGER_ADDR=`make get-eigen-service-manager-from-deploy`
+
+# Deploy trigger and submission contracts
+forge script ./script/Deploy.s.sol ${SERVICE_MANAGER_ADDR} --sig "run(string)" --rpc-url http://localhost:8545 --broadcast
+
+# Verify trigger contract address (optional)
+make get-trigger-from-deploy
+```
+
+3. Deploy WAVS Service:
+```bash
+# Build service configuration
+COMPONENT_FILENAME=eth_price_oracle.wasm sh ./script/build_service.sh
+
+# Deploy service to WAVS
+SERVICE_CONFIG_FILE=.docker/service.json make deploy-service
+```
+
+4. Trigger and Test Service:
+```bash
+# Set required environment variables
+export COIN_MARKET_CAP_ID="your_input"
+export SERVICE_TRIGGER_ADDR=`make get-trigger-from-deploy`
+
+# Execute trigger
+forge script ./script/Trigger.s.sol ${SERVICE_TRIGGER_ADDR} ${COIN_MARKET_CAP_ID} --sig "run(string,string)" --rpc-url http://localhost:8545 --broadcast -v 4
+
+# Wait 10-15 seconds for processing
+# Then check the result
+make show-result
+```
```

Comment:
Again, this is a bit to example specific.

In order to test the service, the user needs to write a script to trigger the event emission.